core/src/main/java/com/sun/mail/util/WriteTimeoutSocket.java

```diff
@@ -419,4 +456,20 @@ public synchronized void write(byte[] bs, int off, int len)
     public void close() throws IOException {
 	os.close();
     }
+
+    private String getTimeoutTaskResult(ScheduledFuture<String> sf)  {
+        boolean wasInterrupted = Thread.interrupted();
+        try {
+            return sf.get();
+        } catch (Exception e) {
+            if (e instanceof InterruptedException) {
+                wasInterrupted = true;
+            }
+            return String.format("%s. %s", CANNOT_GET_TIMEOUT_TASK_RESULT_MESSAGE, e.getMessage());
```

Comment:
@jmehrens thanks again for the review! No problem, I'll change it. However, I'm not sure if I got properly your idea. If we have 3 catch blocks, we will have something like:
```
    private String getTimeoutTaskResult(ScheduledFuture<String> sf)  {
        boolean wasInterrupted = Thread.interrupted();
        String exceptionMessage = null;
        try {
            return sf.get();
        } catch (CancellationException e) {
            exceptionMessage = e.toString();
        } catch (InterruptedException e) {
            exceptionMessage = e.toString();
            wasInterrupted = true;
        } catch (ExecutionException e) {
            exceptionMessage = e.toString();
        } catch (Exception e) {
            exceptionMessage = e.toString();
        } finally {
            if (wasInterrupted) {
                Thread.currentThread().interrupt();
            }
        }

        return String.format("%s. %s", CANNOT_GET_TIMEOUT_TASK_RESULT_MESSAGE, exceptionMessage);
    }
```
But what is the difference if eventually, we have e.toString() for all cases except InterruptedException (where we set a boolean variable)? Also, I think we should keep Exception, just in case something goes wrong it won't impact the execution, what do you think?

Reply:
fixed